PlayCover/AppInstaller/Binary Patcher/Operations.swift

```diff
@@ -0,0 +1,208 @@
+
+        let occupant = binary.subdata(with: NSRange(location: Int(macho.header.sizeofcmds + macho.offset + macho.size),
+                                                    length: Int(length + padding)))
+
+        if occupant[0] != 0 {
+            print("Cannot inject payload into \(dylibPath) because there is no room")
+            return false
+        }
+
+        print("Inserting a \(LC(type)) command for architecture: \(macho.header.cputype)")
+
+        var command = dylib_command()
+        var dylib = dylib()
+        dylib.name.offset = UInt32(MemoryLayout.size(ofValue: dylib_command.self))
+        dylib.timestamp = 2
+        dylib.current_version = 0
+        dylib.compatibility_version = 0
+        command.cmd = type
+        command.dylib = dylib
+        command.cmdsize = UInt32(length + padding)
+
+        var zeroByte: UInt = 0
+        let commandData = NSMutableData(data: Data())
+        commandData.append(&command, length: MemoryLayout.size(ofValue: dylib_command.self))
+        commandData.append(dylibPath.data(using: .ascii)!)
+        commandData.append(&zeroByte, length: Int(padding))
+
+        binary.replaceBytes(in: NSRange(location: Int(macho.offset + macho.header.sizeofcmds + macho.size),
+                                        length: commandData.length), withBytes: nil, length: 0)
+
+        binary.replaceBytes(in: NSRange(location: Int(lastOffset),
+                                        length: 0), withBytes: commandData.bytes, length: commandData.length)
+
+        macho.header.ncmds += 1
+        macho.header.sizeofcmds += command.cmdsize
+
+        binary.replaceBytes(in: NSRange(location: Int(macho.offset),
+                                        length: MemoryLayout.size(ofValue: macho.header)), withBytes: &macho.header)
+
+        return true
+    }
+
+    @discardableResult
+    public static func removeLoadEntryFromBinary(_ binary: inout NSMutableData,
+                                                 _ macho: inout thin_header,
+                                                 _ payload: String) -> Bool {
+        var offset: Int = Int(macho.offset + macho.size)
+
+        var num: UInt32 = 0
+        var cumulativeSize: UInt32 = 0
+        var removedOrdinal: Int32 = -1
+
+        for index in 0..<macho.header.ncmds {
+            if offset >= binary.length || offset > macho.offset + macho.size + macho.header.sizeofcmds {
+                break
+            }
+
+            let cmdData = Data(bytes: binary[offset..<offset+4].base.bytes, count: 4)
+            let cmd = cmdData.withUnsafeBytes { $0.load(as: UInt32.self) }
+
+            let sizeData = Data(bytes: binary[offset+4..<offset+8].base
+                .bytes, count: 4)
+            let size = sizeData.withUnsafeBytes { $0.load(as: UInt32.self) }
+
+            switch cmd {
+            case LC_REEXPORT_DYLIB,
+            LC_LOAD_UPWARD_DYLIB,
+            LC_LOAD_WEAK_DYLIB,
+            UInt32(LC_LOAD_DYLIB):
```

Comment:
why the heck aren't these constants beginning with `LC_` of the same type?

Reply:
Ask Apple, these constants are defined in the MachO.loader framework